wdl2cwl/tests/cwl_files/blast.cwl

```diff
@@ -0,0 +1,335 @@
+    - id: penalty
+      default: -2
+      type: int
+    - id: strand
+      default: both
+      type: string
+    - id: gapopen
+      default: 0
+      type: int
+    - id: gapextend
+      default: 0
+      type: int
+    - id: dust
+      default: "'20 64 1'"
+      type: string
+    - id: max_hsps
+      type:
+        - int
+        - 'null'
+    - id: tasks
+      default: megablast
+      type: string
+    - id: taxids
+      type:
+        - string
+        - 'null'
+    - id: negative_taxids
+      type:
+        - string
+        - 'null'
+    - id: lcase_masking
+      default: false
+      type: boolean
+  baseCommand:
+    - bash
+    - script.bash
+  outputs:
+    - id: out
+      type: File
+      outputBinding:
+        glob: $(inputs.Outfile)
+
+- class: CommandLineTool
+  id: runblastp
+  requirements:
+    - class: InitialWorkDirRequirement
+      listing:
+        - entryname: script.bash
+          entry: |4
+              set -e
+              blastp -db "$(inputs.Fname)" \
+              -query $(inputs.Queryfa) \
+              -outfmt $(inputs.outfmt) \
+              -out     $(inputs.Outfile) \
+              -max_target_seqs $(inputs.max_target_seqs) \
+              -comp_based_stats $(inputs.comp_based_stats) \
+              -evalue $(inputs.evalue) \
+              -word_size $(inputs.word_size) \
+              -matrix   $(inputs.matrix) \
+              -seg     $(inputs.seg) \
+              -gapopen $(inputs.gapopen) \
+              -gapextend $(inputs.gapextend) \
+              -num_threads $(inputs.threads) \
+              $(true='-lcase_masking' false='' inputs.lcase_masking) $("-max_hsps "+inputs.max_hsps) $("-taxids " +inputs.taxids) $("-negative_taxids " +inputs.negative_taxids) \
+  hints:
+    - class: DockerRequirement
+    - class: ResourceRequirement
+      coresMin: 8
+      ramMin: 16384
+  inputs:
+    - id: docker
+      type: string
+    - id: Queryfa
+      type: File
+    - id: Fname
+      type: string
+    - id: outfmt
+      type: string
+    - id: Outfile
+      type: string
+    - id: evalue
+      type: float
+    - id: threads
+      type: int
+    - id: max_target_seqs
+      default: 100
+      type: int
+    - id: word_size
+      default: 6
+      type: int
+    - id: seg
+      default: no
+      type: string
+    - id: comp_based_stats
+      default: '2'
+      type: string
+    - id: matrix
+      default: BLOSUM62
+      type: string
+    - id: gapopen
+      default: 11
+      type: int
+    - id: gapextend
+      default: 1
+      type: int
+    - id: max_hsps
+      type:
+        - int
+        - 'null'
+    - id: taxids
+      type:
+        - string
+        - 'null'
+    - id: negative_taxids
+      type:
+        - string
+        - 'null'
+    - id: lcase_masking
+      default: false
+      type: boolean
+  baseCommand:
+    - bash
+    - script.bash
+  outputs:
+    - id: out
+      type: File
+      outputBinding:
+        glob: $(inputs.Outfile)
```

Comment:
@mr-c , I tried to manually partially translate the `blast.wdl` file. Could you take a look at the file above and see if it's going in the right direction here, please?

There are a couple TODO's where I am not sure how to translate something from WDL, :pray: 

I used two tasks only, translating them both into `CommandLineTool`'s. Also used that `$graph` trick to pack multiple processes in the same file. Finally, added `Workflow` named `main` for the `blast.wdl` workflow definition. The `Workflow` uses conditional to pick the first non null (i.e. `select_first` and `if`'s from WDL).